src/jpeg/decoder.rs

```diff
@@ -134,3 +132,23 @@ impl ImageError {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::cmyk_to_rgb;
+
+    #[test]
+    fn cymk_to_rgb_correct() {
+        // Based on R = 255 * (1-C/255) * (1-K/255)
+        for c in 0..255 {
+            for k in 0..255 {
+                let r = (255.0 - f32::from(c)) * (255.0 - f32::from(k)) / 255.0;
+                let r_u8 = r as u8;
+                let convert_r = cmyk_to_rgb(&[c, 0, 0, k])[0];
```

Comment:
What's the use of this test? It tests the implementation against itself. I would have expected some pre-computed reference table of values that cover combinations of color channels, most importantly the Cyan Magenta and Yellow primaries that are the point of cmyk.

Reply:
The combinations don't affect the output: R only depends on C and K, B depends on M and K, and G depends on Y and K - the calculation is the same for each value. The point of the test is to compute the correct conversion of C and K to R with higher precision, and ensure that the function produces the correct value.

Reply:
Copy & Paste errors happen and it would be useful if they were caught by a test.

Reply:
> The point of the test is to compute the correct conversion of C and K to R with higher precision, and ensure that the function produces the correct value.

If the point is precision then this is the wrong test. `r as u8` does not round, it truncates. The more restrictive you make the test, the more surer you need to be to be testing for the correct properties. I would have accepted a test for just the six primaries of RGB and CMY as well, *since that is was the PR claims to fix*.

Reply:
That's fair - I added some more tests to validate specific colors.